easybuild/easyblocks/p/petsc.py

```diff
@@ -79,6 +82,22 @@ def configure_step(self):
         Configure procedure is much more concise for older versions (< v3).
         """
         if LooseVersion(self.version) >= LooseVersion("3"):
+            # make the install dir first if we are doing a download install, then keep it for the rest of the way
+            if self.cfg["download_packages"] or self.cfg["download_packages_static"] or self.cfg["download_packages_shared"]:
+                self.log.info("Creating the installation directory before the configure.")
+                self.make_installdir()
+                self.cfg["keeppreviousinstall"] = True
+                if self.cfg["download_packages"]:
+                    for p in self.cfg["download_packages"]:
+                        self.cfg.update('configopts', '--with-download-%s=1' % p)
+                if self.cfg["download_packages_static"]:
+                    for p in self.cfg["download_packages_static"]:
+                        self.cfg.update('configopts', '--with-download-%s=1' % p)
+                        self.cfg.update('configopts', '--with-download-%s-shared=0' % p)
+                if self.cfg["download_packages_shared"]:
+                    for p in self.cfg["download_packages_shared"]:
+                        self.cfg.update('configopts', '--with-download-%s=1' % p)
```

Comment:
Done. I also added a "set" directive to make sure that if dependencies are listed multiple times, they are not included in the configure option multiple times.